app/src/main/java/com/github/sgueissa/bootcamp/FirebaseBootcamp.java

```diff
@@ -0,0 +1,57 @@
+package com.github.sgueissa.bootcamp;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.TextView;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import java.util.concurrent.CompletableFuture;
+
+public class FirebaseBootcamp extends AppCompatActivity {
+
+    private DatabaseReference db = FirebaseDatabase.getInstance().getReference();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_firebase_bootcamp);
+    }
+
+    public void get(View view){
+        TextView email = (TextView) findViewById(R.id.email_text);
+        TextView phone = (TextView) findViewById(R.id.phone_text);
+        CompletableFuture<String> future = new CompletableFuture<>();
+        db.child(phone.getText().toString()).get().addOnSuccessListener(new OnSuccessListener<DataSnapshot>() {
+            @Override
+            public void onSuccess(DataSnapshot dataSnapshot) {
+                if (dataSnapshot.getValue() == null) future.completeExceptionally(new NoSuchFieldException());
+                else future.complete(dataSnapshot.getValue().toString());
+            }
+        }).addOnFailureListener(
+                new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        future.completeExceptionally(e);
+                    }
+                }
+        );
+        future.thenAccept(
+                value -> email.setText(value)
+        );
+    }
+
+    public void set(View view){
+        String email = ((TextView) findViewById(R.id.email_text)).getText().toString();
+        String phone = ((TextView) findViewById(R.id.phone_text)).getText().toString();
```

Comment:
Is it possible to save ``(TextView) findViewById(R.id.email_text)`` in a constant or method somehow? For example in line 29/30 we are doing the same thing. For now it's not that important but for later there might be a cleaner way if elements get reused a lot.

Reply:
I pushed the changes in commit [43f7f86]. I replaced findViewById instances by binding classes (which were not included in the bootcamp). They are not yet needed for the tests.